pkg/odo/genericclioptions/resolve.go

```diff
@@ -27,15 +27,12 @@ func (o *internalCxt) resolveProjectAndNamespace(cmdline cmdline.Cmdline, config
 		}
 		namespace = projectFlag
 	} else {
-		namespace = configProvider.GetNamespace()
```

Comment:
You can remove configProvider argument from the function signature. Can you also remove the check on L24?
```go
// do not error out when the user is running `odo project`
		if err != nil {
			if cmdline.GetParentName() != "project" {
				return err
			}
		}
```